hummingbot/strategy/pure_market_making/inventory_cost_price_delegate.py

```diff
@@ -0,0 +1,68 @@
+from decimal import Decimal, InvalidOperation
+
+from hummingbot.core.event.events import OrderFilledEvent, TradeType
+from hummingbot.exceptions import NoPrice
+from hummingbot.model.inventory_cost import InventoryCost
+from hummingbot.model.sql_connection_manager import SQLConnectionManager
+
+
+class InventoryCostPriceDelegate:
+    def __init__(self, sql: SQLConnectionManager, trading_pair: str) -> None:
+        self.base_asset, self.quote_asset = trading_pair.split("-")
+        self._session = sql.get_shared_session()
+
+    @property
+    def ready(self) -> bool:
+        return True
+
+    def get_price(self) -> Decimal:
+        record = InventoryCost.get_record(
+            self._session, self.base_asset, self.quote_asset
+        )
+        try:
+            price = record.quote_volume / record.base_volume
+        except (ZeroDivisionError, InvalidOperation, AttributeError):
+            raise NoPrice("Inventory cost delegate does not have price cost data yet")
```

Comment:
I think it'd be easier to know which conditions exactly causing NoPrice, and it'd be better to use None to handle this rather than exception. e.g.
```python
if record.base_volume <= s_decimal_0 or record.base_volume is None:
    return None
```
InvalidOperation or AttributeError will hide other problem and make it. difficult to debug later on.

Reply:
In general, a function must return only one type. It's a common anti-pattern described here:  https://docs.quantifiedcode.com/python-anti-patterns/maintainability/returning_more_than_one_variable_type_from_function_call.html

For these exceptions, all of them are appear when price data is not ready yet (for different reasons). For debug purposes, we can add info on what exact exception happened:

```python
except (ZeroDivisionError, InvalidOperation, AttributeError) as e:
    raise NoPrice(f"Inventory cost delegate does not have price cost data yet: {str(e)}")
```

Reply:
I, personally, don't like catching exceptions for expected None value, it makes code harder to read (many try-excepts and nested ones). Moreover, our practice so far has been returning None is acceptable (with hint `-> Optional[]`), it'd be easy to maintain if we follow this (but sure when there is clear benefit in accepting new coding practice we should consider).

Reply:
Refactored to return None.